src/options/components/navigation/Nav.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import styles from './styles.css'
+
+import NavLink from './NavLink'
+
+const Nav = ({ routes, currentLocation, state }) => {
+    function buildRoutes() {
```

Comment:
Instead of putting this as a function inside a view, do this type of state derivation (mapping state to views) in the container. You can probably move `buildRoutes` to a container method called `renderNavLinks` which does the same `routes.map` call (but it will probably be `this.props.routes`). 

Then you can have this `Nav` view have a special prop `children` (instead of `routes`), which it just renders out, like:

```jsx
const Nav = ({ children }) => (
  <nav className={styles.root}>
    <div className={styles.icon_div}>
      <img src="/img/worldbrain-logo.png" className={styles.icon} />
    </div>
    <ul className={styles.nav}>{children}</ul>
  </nav>
)
```